Vizzuality/heco-invest
# Add open call cards,  open call search page

## Description

**In this PR:**
- Added an `OpenCallCard` component
  _Based on the existing `ProfileCard` & `ProjectCard`_ 
  - Added its respective Storybook story  
    _also added an investor.json and an openCall.json mock files to help with Storybook stories_
- `open-call-service` changes:  
  - Added a new hook to retrieve the list of open calls from the backend
    _with pagination, filtering, sorting and searching support_
- `DiscoverPageLayout` changes:  
  - Small refactoring to simplify how we load projects/pds/investors  
  - Added support for open calls  
    _including navigation badge with the correct count_
- `discover/open-calls` page changes:  
  - Populated the page with a list of open calls  
    _using the new `OpenCallCard` component_  
    - Pagination is implemented and working
    - Sorting is implemented
      _Note: sorting by `instrument_type` sends the query param to the endpoint, but the backend doesn't have it implemented yet_
    - Filtering is implemented
      _~Note: the frontend sends the query params to the endpoint, but it's not all implemented in the backend~_
    - Searching by keywords is implemented  
- `SortingButtons` component and `useSortingByOptions` hook changes: 
  - ~Added a `instrument_type` to the sorting buttons types~  
  - Refactored the hook to specify the target page we need the sorting options for
    _eg: if we pass `projects` it'll return the sorting options for projects and so on. By default it'll only return the `name` and `date` options_. 
- Added `Head` titles
  _to the discover pages_


## Testing instructions

Verify that:  
- `discover/open-calls` loads correctly  
  - The correct number of search results appears in the navigation badges  
  - Non-ongoing open calls have a ribbon with the text "Open call ended"  on their cards  
  - The information displayed in the cards is correct  
  - The investor info / tags displayed in the cards is correct  
- Pagination works as expected  
- Sorting works as expected  
  **Note:** Sorting by `instrument_type` doesn't seem to be implemented in the backend, but we can check for correct query params in the url/dev tools
- Searching works as expected  
- Filtering: check that the correct query params are sent in the url/dev tools
- Sorting: check that the correct query params are sent in the url/dev tools

## Tracking

[LET-888](https://vizzuality.atlassian.net/browse/LET-888)

## Screenshot

<img width="1680" alt="Screenshot 2022-08-25 at 13 45 07" src="https://user-images.githubusercontent.com/6273795/186668117-06a46a9a-c276-47da-9bff-7d45204a2afc.png">
